rust/kernel/sync/locked_by.rs

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! A wrapper for data protected by a lock that does not wrap it.
+
+use super::{Guard, Lock};
+use core::{cell::UnsafeCell, ops::Deref, ptr};
+
+/// Allows access to some data to be serialised by a lock that does not wrap it.
+///
+/// In most cases, data protected by a lock is wrapped by the appropriate lock type. [`LockedBy`]
```

Comment:
Let's try to drive users a bit harder into existing APIs to help avoiding them using standalone locks if possible; something like:

```suggestion
/// In most cases, data protected by a lock should be wrapped by the appropriate lock type,
/// e.g. [`Mutex`], instead of using an standalone lock. [`LockedBy`]
```

Reply:
Sure.